cpp/src/gandiva/gandiva_aliases.h

```diff
@@ -19,8 +19,13 @@
 #define GANDIVA_ALIASES_H
 
 #include <memory>
+#include <string>
+#include <unordered_set>
 #include <vector>
 
+#include <boost/functional/hash.hpp>
```

Comment:
why here ? this file is included in any places - can we move it to only the files where it's required ?